### PR TITLE
Add test in test_azure.py

### PR DESCRIPTION
**Pull request by KsenijaS**
Add additional test in test_azure.py to vet the expected behavior of suppressing error messages from _mount_cb_. 
This is addressing my previous PR https://github.com/canonical/cloud-init/pull/2134

### CODE REVIEW

File: tests/unittests/sources/test_azure.py [hunk 1]
```diff
@@ -2245,38 +2245,70 @@ def partitions_on_device(devpath):
...
+                "/dev/sda": {
+                    "partitions": {
+                        "/dev/fake0": {"num": 1, "fs": "ntfs", "files": []}
+                    }
+                }
+            }
+        )
+
+        log_msg = (
+            "Failed to mount device: '/dev/fake0' with type: "
```
Comment (cjp256) [inline tests/unittests/sources/test_azure.py:2279]: Consider simplifying this string a bit to ensure it's not as error-prone. e.g. "Failed to mount device"

File: tests/unittests/sources/test_azure.py [hunk 2]
```diff
@@ -2240,38 +2240,71 @@ def partitions_on_device(devpath):
...
+        )
+
+        m_subp.side_effect = subp.ProcessExecutionError(
+            "", "unknown filesystem type 'ntfs'"
+        )
+
+        dsaz.can_dev_be_reformatted("/dev/sda", preserve_ntfs=False)
+        self.assertNotIn(log_msg, self.logs.getvalue())
+
+    def mount(self, bypath):
```
Comment (cjp256) [inline tests/unittests/sources/test_azure.py:2290]: suggest renaming to _domock_mount_cb() or something to that effect so it doesn't read like we're mounting

**Review by cjp256 — COMMENTED** [review]
lgtm with these changes

**Comment by holmanb** [comment]
@KsenijaS Thanks for this contribution. Can you please create a proposed commit message for us to squash merge with?

**Review by TheRealFalcon — APPROVED** [review]
LGTM!

**Review by TheRealFalcon — APPROVED** [review]
LGTM!

**Pull request merged**